### PR TITLE
Cache the result of source_of (#172)

### PR DESCRIPTION
**Pull request by ShikChen**
This change is [Reviewable](https://reviewable.io/reviews/peter50216/pwntools-ruby/174)

### CODE REVIEW

File: lib/pwnlib/logger.rb [hunk 1]
```diff
@@ -130,7 +130,11 @@ def add(severity, message = nil, progname = nil)
       end
 
       def source_of(path, line_number)
-        File.open(path) { |f| LoggerType.expression_at(f, line_number) }
+        @source_of_file_cache = Hash.new do |h, key|
```
Comment (peter50216) [inline lib/pwnlib/logger.rb]: This doesn't cache :(

Comment (ShikChen) [reply lib/pwnlib/logger.rb]: Moved to initialize().

**Comment by david942j** [comment]
Plz fix the test, apparently File#close has different behavior before Ruby 2.5

**Comment by ShikChen** [comment]
> Plz fix the test, apparently File#close has different behavior before Ruby 2.5

The problem is in Tempfile::create. [Ruby <=2.4](https://ruby-doc.org/stdlib-2.4.0/libdoc/tempfile/rdoc/Tempfile.html#method-c-create) does not check if the file is already removed before calling File::unlink, and [Ruby 2.5](https://ruby-doc.org/stdlib-2.5.0/libdoc/tempfile/rdoc/Tempfile.html#method-c-create) fixed this issue.

I will change the code to fix the test.

**Pull request merged**